mw/hook.d.ts

```diff
@@ -41,30 +41,30 @@
  *
  * @see https://doc.wikimedia.org/mediawiki-core/master/js/#!/api/mw.hook
  */
-interface Hook {
+interface Hook<T extends any[] = any[]> {
     /**
      * Register a hook handler
      *
      * @param {...Function} handler Function to bind.
      * @chainable
      */
-    add(...handler: Array<(...args: any[]) => any>): Hook;
+    add(...handler: ((...args: T) => any)[]): this;
 
     /**
      * Run a hook.
      *
      * @param {*} data
      * @chainable
      */
-    fire(data?: any): Hook;
+    fire(...data: T): this;
 
     /**
      * Unregister a hook handler
      *
      * @param {...Function} handler Function to unbind.
      * @chainable
      */
-    remove(handler: (...args: any[]) => any): Hook;
+    remove(handler: (...args: T) => any): this;
```

Comment:
According to the source code, we can specify multiple handlers for removal - let's make this varargs as well.